include/natalie/integer_object.hpp

```diff
@@ -78,7 +78,7 @@ class IntegerObject : public Object {
         if (integer->is_bignum())
             env->raise("RangeError", "bignum too big to convert into `{}'", typeid(T).name());
         const auto result = integer->to_nat_int_t();
-        if (!std::numeric_limits<T>::is_signed && result < 0)
+        if (std::numeric_limits<T>::is_signed && result < 0)
```

Comment:
This change is incorrect. It might be a bit confusing, since we check for the type being signed and reverse that result (there is no `is_unsigned` property in `std::numeric_limits`).

Easiest way to test this:
```sh
ruby -e 'p File.open("/dev/zero").read(-1)'
bin/natalie -e 'p File.open("/dev/zero").read(-1)'
```
Both these lines (with Natalie master) throw `-e:1:in `read': negative length -1 given (ArgumentError)`

With the above fix, Natalie throws `-e:1:in `read': integer -1 too small to convert to `size_t' (RangeError)`, which is still correct, but not the expected error.

Reply:
Thanks. This PR is draft because I am still fiddling with it.